oauth2client/transport/httplib2.py

```diff
@@ -44,32 +44,31 @@ def get_http_object(*args, **kwargs):
     return httplib2.Http(*args, **kwargs)
 
 
-def inject_credentials(credentials, http, refresh_status_codes):
-    """Prepares an HTTP object's request method for auth.
-
-    Wraps HTTP requests with logic to catch auth failures (typically
-    identified via a 401 status code). In the event of failure, tries
-    to refresh the token used and then retry the original request.
-
-    Args:
-        credentials: Credentials, the credentials used to identify
-                     the authenticated user.
-        http: httplib2.Http, an http object to be used to make
-              auth requests.
+class AuthorizedHttp(object):
+    """A httplib2.Http-like object that provides credentials to requests.
+
+    A new class is used because you can't create a new OAuth subclass of
```

Comment:
This class makes clear why the class has to exist. Why does the class have to be library-public?